Feature/nre/3 secured authentification form

nicolasrll opened the pull request:
#3

nicolasrll committed:
feat #3 : CSRF token implemented

issue #3

nicolasrll committed:
feat #3 : csrfTokenCheck function moved to be in DefaultControllerAbstract

issue #3

fmoy commented on config/conf.php:
```diff
@@ -1,6 +1,6 @@
-    define('DBNAME' /*, database name value*/);
-    define('DBUSER' /*, login value*/);
-    define('DBPASSWORD' /*, password value*/);
+    define('DSN', 'mysql:host=localhost');
```
ne pas versionner ce fichier utiliser le .dist

fmoy commented on core/DefaultControllerAbstract.php:
```diff
@@ -54,5 +54,11 @@ public function isSubmited(string $arg): bool
         return true;
     }
+
+    public function hasCSRFToken()
```
setTokenCSRF / generateTokenCSRF

fmoy replied:
typer le retour

fmoy commented on src/Controller/AuthentificationController.php:
```diff
@@ -24,10 +26,14 @@ public function loginAction()
         if ($this->isSubmited('authentification')) {
             $formValues = $this->getFormValues('authentification');
+
+            //$this->csrfTokenCheck($formValues['token']);
```
virer les commentaires

fmoy commented on src/Controller/AuthentificationController.php:
```diff
@@ -24,10 +26,14 @@ public function loginAction()
     {
         if ($this->isSubmited('authentification')) {
```
j'ajouterais le test sur le token

fmoy commented on src/Controller/AuthentificationController.php:
```diff
@@ -24,10 +26,14 @@ public function loginAction()
 
             if (
-                empty($user)
+                !$this->csrfTokenCheck($formValues['token'])
```
retirer le check sur le role (la redirection s'en chargera) retirer le check sur le token, le faire plus tot, et ne plus utiliser cette methode, utiliser celle de l'indexAction();

fmoy commented on src/Controller/AuthentificationController.php:
```diff
@@ -45,11 +51,25 @@ public function loginAction()
         );
     }
 
+    public function csrfTokenCheck($formTokenValue): bool
```
verifyTokenCsrf() / checkTokenCsrf()

fmoy commented on template/authentification-admin.html.twig:
```diff
@@ -29,6 +29,7 @@
                                 <label for="password">Mot de passe</label>
                                 <input class="form-control" type="password" id="password" name="authentification[password]" required>
                             </div>
+                            <input type="hidden" name="authentification[token]" value="{{ session.token }}">
```
à voir pour gérer plusieurs token (un par formulaire pour éviter les conflits)

fmoy commented on .gitignore:
```diff
@@ -0,0 +1,4 @@
+.idea/
```
revois mes anciennes remarques

nicolasrll committed:
feat #3 : modification following the last meeting and refactoring

Renaming of hasCSRFToken () to generateTokenCSRF ()
Improvement of the message returned after verification of the token
Name of the token in the modified form
Function indexAction in AuthentificationController refactored

issue #3

nicolasrll committed:
feat #2 : application of pull requst messages

Correction done in construct of DefaultControllerAbstract
Correction on return type on indexAction in AuthentificationController
File in .dist created to ignore conf.php
